torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,328 @@
+        Returns:
+            tuple: A 4-tuple with ``(img_left, img_right, disparity, valid_mask)``.
+            The disparity is a numpy array of shape (1, H, W) and the images are PIL images.
+            ``valid_mask`` is implicitly ``None`` if the ``transforms`` parameter does not
+            generate a valid mask.
+            Both ``disparity`` and ``valid_mask`` are ``None`` if the dataset split is test.
+        """
+        return super().__getitem__(index)
+
+
+class Kitti2015Stereo(StereoMatchingDataset):
+    """Kitti dataset from the `2015 <http://www.cvlibs.net/datasets/kitti/eval_scene_flow.php>`_ stereo evaluation benchmark.
+    The dataset is expected to have the following structure: ::
+        root
+            Kitti2015
+                testing
+                    image_2
+                        img1.png
+                        img2.png
+                        ...
+                    image_3
+                        img1.png
+                        img2.png
+                        ...
+                training
+                    image_2
+                        img1.png
+                        img2.png
+                        ...
+                    image_3
+                        img1.png
+                        img2.png
+                        ...
+                    disp_occ_0
+                        img1.png
+                        img2.png
+                        ...
+                    disp_occ_1
+                        img1.png
+                        img2.png
+                        ...
+                    calib
+    Args:
+        root (string): Root directory where Kitti2015 is located.
+        split (string, optional): The dataset split of scenes, either "train" (default) or test.
+        transforms (callable, optional): A function/transform that takes in a sample and returns a transformed version.
+    """
+
+    def __init__(self, root: str, split: str = "train", transforms: Optional[Callable] = None):
+        super().__init__(root, transforms)
+        self._has_built_in_disparity_mask = True
+
+        verify_str_arg(split, "split", valid_values=("train", "test"))
+
+        root = Path(root) / "Kitti2015" / (split + "ing")
+        left_img_pattern = str(root / "image_2" / "*.png")
+        right_img_pattern = str(root / "image_3" / "*.png")
+        self._images += self._scan_pairs(left_img_pattern, right_img_pattern)
+
+        if split == "train":
+            left_disparity_pattern = str(root / "disp_occ_0" / "*.png")
+            right_disparity_pattern = str(root / "disp_occ_1" / "*.png")
+            self._disparities += self._scan_pairs(left_disparity_pattern, right_disparity_pattern)
+        else:
+            self._disparities = list(("", "") for _ in self._images)
+
+    def _read_disparity(self, file_path: str) -> Tuple:
+        # test split has no disparity maps
+        if not os.path.exists(file_path):
+            return None, None
```

Comment:
I would rather suggest to do `""` instead of `None`  as we can check with `if not file_path`. Setting it to a list of `None` would start throwing mypy errors for:

```python3
if paths_right_pattern:
            right_paths = sorted(glob(paths_right_pattern))
else:
            right_paths = list(None for _ in left_paths)
```

So, to address both this issue and https://github.com/pytorch/vision/pull/6311#discussion_r931259039 I would personally store `""` when no pattern is specified and switch to a `if not file_path` check.

Reply:
> mypy errors

That's why in https://github.com/pytorch/vision/pull/6311/files#r931262310 I suggested to keep type annotations to the bare minimum (feel free to entirely remove them as well, I won't ask to bring them back).

As much as possible, we should avoid letting mypy be in the way of writing Pythonic code (it's a hard task, I would agree).